source/extensions/access_loggers/common/access_log_base.cc

```diff
@@ -0,0 +1,33 @@
+#include "extensions/access_loggers/common/access_log_base.h"
+
+#include "common/http/header_map_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace AccessLoggers {
+namespace Common {
+
+void ImplBase::log(const Http::HeaderMap* request_headers, const Http::HeaderMap* response_headers,
+                   const Http::HeaderMap* response_trailers,
+                   const StreamInfo::StreamInfo& stream_info) {
+  const Http::HeaderMapImpl empty_headers;
```

Comment:
I'm OK with this but it might be worth it to use a lazy singleton; you just need it to be a const pointer, rather than a non-const value. The reason is that (I believe) HeaderMapImpl is not a trivial ctor and it is better not to have to construct it on every log request.

Reply:
Wrapped this in a ConstSingleton instead.